### PR TITLE
#18233 integrating app into the translate actionlet

### PR DESCRIPTION
**Pull request by jdotcms**
Integrating the translation actionlet with the apps

Created tests for the secret getter, the translation functionality is already tested in `GoogleTranslationServiceTest`.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/translate/GoogleTranslationService.java [hunk 1]
```diff
@@ -121,7 +135,47 @@ public void setServiceParameters(List<ServiceParameter> params) {
...
+        final HttpServletRequest request = HttpServletRequestThreadLocal.INSTANCE.getRequest();
+        if (null != request) {
+            host = Try.of(()->WebAPILocator.getHostWebAPI().getCurrentHostNoThrow(request)).getOrNull();
+        }
+
+        if (null == host) {
+            host = Try.of(() -> APILocator.getHostAPI().findDefaultHost(APILocator.systemUser(), false)).getOrNull();
+        }
+
+        return null == host? APILocator.systemHost(): host;
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/translate/GoogleTranslationService.java:178]: I think we are missing some test here

**Review by freddyucv — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotcms/translate/TranslationService.java [hunk 2]
```diff
@@ -1,11 +1,13 @@
 package com.dotcms.translate;
 
+import com.dotmarketing.beans.Host;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/translate/TranslationService.java:3]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.beans.Host'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5384201)

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/translate/TranslationService.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/translate/TranslationService.java:3]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.beans.Host'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5384201)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/translate/TranslationService.java:3]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.beans.Host'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5384201)

File: dotCMS/src/main/java/com/dotcms/translate/GoogleTranslationService.java [hunk 3]
```diff
@@ -1,40 +1,54 @@
...
     private String serviceUrl = Config.getStringProperty("GOOGLE_TRANSLATE_SERVICE_BASE_URL",
         BASE_URL);
     private String apiKey;
     private List<ServiceParameter> params;
+    public static final String GOOGLE_TRANSLATE_APP_CONFIG_KEY = "googleTranslate-config";
+    public static final String API_KEY_VAR = "apiKey";
 
     public GoogleTranslationService() {
-        this(Config.getStringProperty("GOOGLE_TRANSLATE_SERVICE_API_KEY", ""), new JSONTool(), new ApiProvider());
+        this(Config.getStringProperty("GOOGLE_TRANSLATE_SERVICE_API_KEY", StringPool.BLANK), new JSONTool(), new ApiProvider());
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/translate/GoogleTranslationService.java:51]: Codacy found an issue: [The String literal "GOOGLE_TRANSLATE_SERVICE_API_KEY" appears 5 times in this file; the first occurrence is on line 51](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5384201)

File: dotCMS/src/integration-test/java/com/dotcms/translate/GoogleTranslationServiceIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.liferay.portal.model.User;
+import com.liferay.util.StringPool;
+import com.dotmarketing.util.Config;
+import java.util.Collections;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class GoogleTranslationServiceIntegrationTest {
```
Comment (nollymar) [inline dotCMS/src/integration-test/java/com/dotcms/translate/GoogleTranslationServiceIntegrationTest.java:21]: Tests need doc

**Review by jdotcms — COMMENTED** [review]
very good Erick

File: dotCMS/src/integration-test/java/com/dotcms/translate/GoogleTranslationServiceIntegrationTest.java (hunk 4 above)
Comment (erickgonzalez) [reply dotCMS/src/integration-test/java/com/dotcms/translate/GoogleTranslationServiceIntegrationTest.java:25]: done

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/translate/GoogleTranslationServiceIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+    private static User admin;
+    private static GoogleTranslationService translationService;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/translate/GoogleTranslationServiceIntegrationTest.java:28]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5384201)

Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/translate/GoogleTranslationServiceIntegrationTest.java:28]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5384201)

**Pull request merged**